src/compressor.js

```diff
@@ -0,0 +1,146 @@
+define(function (require) {
+	'use strict';
+
+	var p5sound = require('master');
+	var Effect = require('effect');
+  var CustomError = require('errorHandler');
+
+
+  /**
+   * Built on Web Audio Dynamics Compressor Node
+   * https://www.w3.org/TR/webaudio/#the-dynamicscompressornode-interface
```

Comment:
This merits a more details explanation and an example. That could be done in a future PR but let's make sure to provide some information about what a compressor is, and what all these parameters means. It is a difficult concept to explain, but an interactive example should help :)